### ISSUE TITLE
the future of Veles: adoption by the Kaitai community?

### ISSUE THREAD

**Issue by pabs3** [kaitai-io/kaitai_struct#895]
Hi all,

The [Veles](https://github.com/codilime/veles) [project](https://veles.io/) is a binary data analysis and visualization tool based on Kaitai Struct and created by @mkow, @mwkmwkmwk and others from @codilime.

Unfortunately [CodiLime](https://codilime.com/) discontinued their [CodiSec](https://codilime.com/) efforts, retired Veles, discontinued Veles development and archived the Veles repository.

I've always thought Veles was an interesting tool and was sad to see that it got archived.

So I was wondering if the Kaitai Struct community would be interested in having the Veles repository including issues and pull requests transferred to the kaitai-io GitHub organisation and continuing development of Veles.

The Veles codebase is mostly C++ and also some Python. It is using the Qt GUI framework and the CMake build system. I believe the Kaitai Struct community has the knowledge needed to maintain it.

Aside from the open issues and pull requests, the main work needed would be to update to more recent versions of Qt. It also contains C++ code pre-generated from I believe the [Kaitai formats repo](https://github.com/kaitai-io/kaitai_struct_formats), so that could be replaced with building from the source `.ksy` files.

**Comment by KOLANICH** [comment]
I can hypothesize that Hobbits author, @hello-adam, may  be more interested in Veles than we.  His tool lacks near-useless 3d visualization, but 2d-capabilities are more advanced. I guess some Veles functionality can be ported to his tool.

**Comment by hello-adam** [comment]
yeah, Hobbits isn't quite a first-class Kaitai environment yet, but it's getting there. Let me know what capabilities or niceties of Veles you think I should try to port/implement in order to meet your needs!

https://github.com/Mahlet-Inc/hobbits
